read-fonts/src/tables/gsub/closure.rs

```diff
@@ -1033,4 +1050,43 @@ mod tests {
         let nop = compute_closure(&gsub, &glyph_map, &["a", "b", "c"]);
         assert_closure_result!(glyph_map, nop, &["a", "b", "c"]);
     }
+
+    #[test]
+    fn collect_all_features() {
+        let font = FontRef::new(font_test_data::closure::CONTEXTUAL).unwrap();
+        let gsub = font.gsub().unwrap();
+        let ret = gsub.collect_features(None, None, None).unwrap();
+        assert_eq!(ret.len(), 2);
+        assert!(ret.contains(0));
+        assert!(ret.contains(1));
+    }
+
+    #[test]
+    fn collect_all_features_with_feature_filter() {
+        let font = FontRef::new(font_test_data::closure::CONTEXTUAL).unwrap();
+        let gsub = font.gsub().unwrap();
+
+        let mut feature_tags = IntSet::empty();
+        feature_tags.insert(Tag::new(b"SUB5"));
+
+        let ret = gsub
+            .collect_features(None, None, Some(&feature_tags))
+            .unwrap();
+        assert_eq!(ret.len(), 1);
+        assert!(ret.contains(0));
+    }
+
+    #[test]
+    fn collect_all_features_with_script_filter() {
+        let font = FontRef::new(font_test_data::closure::CONTEXTUAL).unwrap();
+        let gsub = font.gsub().unwrap();
+
+        let mut script_tags = IntSet::empty();
+        script_tags.insert(Tag::new(b"LATN"));
+
+        let ret = gsub
+            .collect_features(Some(&script_tags), None, None)
+            .unwrap();
+        assert!(ret.is_empty());
+    }
```

Comment:
Can you also add tests that use inverted sets? Or if you remove the Option as I mentioned above then switch the None's to inverted sets would cover that.